the-collab-lab/tcl-11-smart-shopping-list
# Feature(Issue: #2) : Add the List and AddItem component views

## Description

Resolved issue #2 by adding the `<List />` and `<AddItem />` component views. Added both `react-router-dom` and `node-sass` as project dependencies. Utilized built-in React `<NavLink />` to display active links with custom styling in the Footer component.

## Related Issue

Closes Issue: #2 

## Acceptance Criteria

- Ensure the `react-router-dom` has been added as a project dependency
- Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
- Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![Before Picture](https://user-images.githubusercontent.com/32943879/89444310-88f5f580-d71f-11ea-93be-efd7059be339.png)

### After

![After Picture of List View](https://user-images.githubusercontent.com/32943879/89443954-02d9af00-d71f-11ea-8387-2ff2ce7021a2.png)

![After Picture of AddItem View](https://user-images.githubusercontent.com/32943879/89444135-403e3c80-d71f-11ea-93cf-f5d8d6858de9.png)

## Testing Steps / QA Criteria

- Run `git check out bl-kl-create-view-navigation`
- Run `npm install` to install `react-router-dom` and `node-sass` as project dependencies
- In the browser, visit the home page (http://localhost:3000/). Note, at the bottom of the page, the footer component with present and persistent links to the “list” view and the “add an item” view. Try clicking both links. Whichever view is selected, the corresponding link displays bold text.
